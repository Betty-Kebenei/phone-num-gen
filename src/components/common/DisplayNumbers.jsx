import React from 'react';
import PropTypes from 'prop-types';

const DisplayCurrentGeneration = (props) => {
  const { numbers } = props;

  return (
    <div className="display-current-generation">
      {numbers.map((number, index) => index === 0 ? <span key={number}>{number}</span> : <span key={number}>,{number}</span>)}
    </div>
  )
}
DisplayCurrentGeneration.propTypes = {
  numbers: PropTypes.objectOf.isRequired
}
export default DisplayCurrentGeneration
