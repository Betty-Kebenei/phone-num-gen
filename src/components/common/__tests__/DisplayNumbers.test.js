import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import sinon from 'sinon'
import Adapter from 'enzyme-adapter-react-16';

import DisplayNumbers from '../DisplayNumbers';

configure({ adapter: new Adapter() })

const props = {
  numbers: []
}

describe('DisplayNumbers', () => {
  test('DisplayNumbers component renders correctly', () => {
    const component = renderer.create(<DisplayNumbers  {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('handlePageChange', () => {
    test('active page changes', () => {
      const wrapper = mount(<DisplayNumbers {...props} />);
      const renderedComponent = wrapper.instance();
      expect(wrapper.state('activePage')).toEqual(1);
      sinon.spy(renderedComponent, 'handlePageChange');
      renderedComponent.handlePageChange(2);
      expect(wrapper.state('activePage')).toEqual(2);
    })
  })
})

