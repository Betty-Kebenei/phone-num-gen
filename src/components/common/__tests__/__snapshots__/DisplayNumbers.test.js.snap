// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayNumbers DisplayNumbers component renders correctly 1`] = `
<div>
  <div
    className="display-current-generation"
  />
  <div
    className="paginate-numbers"
  >
    <ul
      className="pagination"
    >
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className=""
          href="#"
        >
          «
        </a>
      </li>
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className=""
          href="#"
        >
          ⟨
        </a>
      </li>
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className=""
          href="#"
        >
          ⟩
        </a>
      </li>
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className=""
          href="#"
        >
          »
        </a>
      </li>
    </ul>
  </div>
</div>
`;
